### DISCUSSION TITLE
How to prevent exposing server IPs in Received header

### DISCUSSION THREAD

**Discussion by iplanwebsites** [postalserver/postal#1964] [Help with using Postal]
While it's fine for most cases, it can exposes some other systems IP adresses. Any way to prevent the API from showing these in the message headers?
A custom Received header get overwritten. I'd be fine with using the Postal Ip if one is required, but ideally I'd like to avoid the otherwise necessary step of creating a proxy just to relay those api calls.

Exemple of a header included. Exposing the IP of the script calling the api.
```
X-Postal-MsgID: TbpZUnJToPhf
Received: from api (fixed-122-190-111-142.tota.net [122.190.111.142]) by Postal with HTTP; Thu, 21 Apr 2022 17:27:02 +0000
Date: Thu, 21 Apr 2022 17:27:02 +0000
```

**Comment by willpower232** [comment]
You'd have to find the right bit of code to remove I believe, I think someone else asked this somewhere...

**Reply by iplanwebsites** [reply]
Looks like it's been discussed a few times and that the change is isolated in one file. For reference:
https://github.com/postalserver/postal/discussions/1471

Line to apply conditional masking / replacement:
https://github.com/postalserver/postal/blob/master/app/models/outgoing_message_prototype.rb#L182

I'd love to add a setting for this but I'm not familiar enough with Ruby.
